libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -548,6 +710,114 @@ impl NodeAPI for Greenlight {
+        let mut client: node::ClnClient = self.get_node_client().await?;
+
+        // We first calculate for each channel the max amount to pay (at the receiver)
+        let mut max_amount_per_channel = self
+            .max_sendable_amount(Some(hex::decode(invoice.payee_pubkey)?), max_hops, last_hop)
+            .await?;
+        info!("send_pay: routes: {:?}", max_amount_per_channel);
+
+        // Calculate the total amount to pay
+        let total_msat: u64 = max_amount_per_channel.iter().map(|m| m.amount_msat).sum();
+
+        // Sort the channels by max amount descending so we can build the route in a way that it
+        // drains the largest channels first
+        max_amount_per_channel.sort_by_key(|m| Reverse(m.amount_msat));
+
+        let amount_to_pay_msat = match invoice.amount_msat {
+            Some(amount) => Ok(amount),
+            None => Err(NodeError::Generic(anyhow!("Invoice has no amount"))),
+        }?;
+
+        if amount_to_pay_msat > total_msat {
+            return Err(NodeError::RouteNotFound(anyhow!(
+                "amount too high, max amount is {} msat",
+                total_msat
+            )));
+        }
+
+        // This is needed in greenlight for the signer to recognize this invoice.
+        client
+            .pre_approve_invoice(PreapproveinvoiceRequest {
+                bolt11: Some(bolt11.clone()),
+            })
+            .await?;
+
+        // We need to allocate a part id for each part that we are sending.
+        let mut part_id = 1;
+        // The total amount we sent. i.e. what the recipient received + fees
+        let mut amount_sent_msat = 0;
+        // The total amount received by the recipient
+        let mut amount_received_msat = 0;
+
+        // The algorithm goes over each channel and drains it until the received amount
+        // equals to the amount to pay defined in the bolt11 invoice.
+        for max in max_amount_per_channel {
+            // calculating the incoming amount for the remaining amount to pay.
+            let left_to_pay_msat = amount_to_pay_msat - amount_received_msat;
+            // Whether we draining the whole channel balance or only what is left to pay
+            let to_pay_msat = std::cmp::min(left_to_pay_msat, max.amount_msat);
+
+            // We convert our payment path to an actual route that can be sent to the node.
+            // This requires calculating the right fees and cltv delta in each hop.
+            let (route, sent_msat) = convert_to_send_pay_route(
+                max.path.clone(),
+                to_pay_msat,
+                invoice.min_final_cltv_expiry_delta,
+            );
+            info!(
+                "send_pay route to pay: {:?}, received_amount = {}",
+                route, to_pay_msat
+            );
+
+            // We send the part using the node API
+            client
+                .send_pay(SendpayRequest {
```

Comment:
re https://github.com/breez/breez-sdk/pull/466#discussion_r1383164929

Reply:
See comment there, I prefer to keep it simple for this first version.